lnrpc+rpcserver: Adding timestamps to RoutingPolicy

ccdle12 opened the pull request:
fixes #2310

This is an early PR. 

This PR adds a `LastUpdate` field to the `RoutingPolicy` message in the `rpc.proto` file and subsequent generated files.

It also adds the `LastUpdate` field in the `rpcserver` and displays a unix timestamp when each party updates their channel policy.

Would appreciate any feedback with the following:
* I'm currently looking for tests to update according to these changes
* Do we also need to remove the LastUpdate field in the `ChannelEdge`?

halseth commented on lnrpc/rpc.proto:
```diff
@@ -1527,6 +1527,7 @@ message RoutingPolicy {
     int64 fee_base_msat = 3 [json_name = "fee_base_msat"];
     int64 fee_rate_milli_msat = 4 [json_name = "fee_rate_milli_msat"];
     bool disabled = 5 [json_name = "disabled"];
+    uint32 last_update = 6 [json_name = "last_update"];
```
protos should be regenerated in this commit.

ccdle12 replied:
fixed!

halseth commented:
@ccdle12 What do you mean by "remove the LastUpdate field in the ChannelEdge?"

cfromknecht commented:
@ccdle12 by chance are you using a different version of the proto compiler? The output formats look quite different

ccdle12 commented:
@cfromknecht My protoc version is currently :

`libprotoc 3.6.1`

Also strangely enough, when I run `gen_protos.sh` it creates a folder `github` in the `lnrpc` directory with all the newly generated protobuf files.

ccdle12 commented:
> @cfromknecht My protoc version is currently :
> 
> `libprotoc 3.6.1`
> 
> Also strangely enough, when I run `gen_protos.sh` it creates a folder `github` in the `lnrpc` directory with all the newly generated protobuf files.

@cfromknecht I was just going through the contribution guidelines and its states the `protoc` version being used is `v3.4.0`, if thats indeed the case as the most up to date recommended version, I can downgrade and then regenerate the files.

wpaulino commented:
@ccdle12 for now `v3.4.0` is the one to use.

ccdle12 commented:
> @ccdle12 for now `v3.4.0` is the one to use.

@wpaulino thanks, the last fixup! adding the deprecated annotation was generated using `v3.4.0`

wpaulino commented:
@ccdle12 I believe the other lnrpc changes weren't generated with `v3.4.0` as well.

halseth commented:
Still looks like wrong proto compiler has been used.

ccdle12 commented:
@halseth thanks, when I run protoc --version:

```
$ protoc --version
$ libprotoc 3.4.0
```

I'll completely remove the binaries and follow the installation instructions again.

halseth commented:
Also check that go-gen-proto and the other proto tools are the correct version

ccdle12 commented:
> Also check that go-gen-proto and the other proto tools are the correct version

Thanks @halseth, I completely removed all project dependencies and reinstalled. It seems like the proto files are being generated per `protoc v3.4.0`. There were updates to `go.mod` and `go.sum` although I'm not sure if those are normal considering the updates.

ccdle12 commented:
> Needs a squash and rebase, and regenerate the protos.
> 
> The `go.*` files should not be modified if done correctly 😀

@halseth 

Thanks for the review and so sorry it's been a while!

I've squashed and rebased according to the updated version of the proto file. 

I've also bumped the number of the `last_update` field in `RoutingPolicy` to 7, in light of the new additional field `max_htlc_msat` in `RoutingPolicy`.

halseth commented:
@ccdle12 Still looks like your proto tools are not the correct version?

ccdle12 commented:
@halseth thanks for catching that, apologies for the oversight, I think I switched version to 2 for a different project. Should be regenerated with version 3.

ccdle12 commented:
@wpaulino @cfromknecht thanks for the review everyone, I'll keep an eye on the PR and keep rebasing until ready to merge

cfromknecht commented:
@ccdle12 needs rebase since #3279 was merged

ccdle12 commented:
> @ccdle12 needs rebase since #3279 was merged

@cfromknecht thanks! rebased

cfromknecht commented:
@ccdle12 :( needs another
